datafusion/proto/src/to_proto.rs

```diff
@@ -1188,7 +1188,28 @@ impl TryFrom<&ScalarValue> for protobuf::ScalarValue {
                 })
             }
 
-            datafusion::scalar::ScalarValue::Time64(v) => {
+            // Since the protos only support Time64 and always interpret it to nanosecond accuracy,
```

Comment:
Question: is there a particular reason why we can't represent these types directly in our serialization format other than backwards compatibility?

Reply:
I don't think so, in fact I would guess we wanted to represent the other types as well, i.e. protos would support `Time32Second`, `Time32Millisecond`, `Time64Microsecond`, `Time64Nanosecond`. I didn't want to change the protos before discussing them with you first. I would suggest the following changes:

`// Contains all valid datafusion scalar type except for
// List
enum PrimitiveScalarType{

    (...)
    INTERVAL_MONTHDAYNANO = 29;

    BINARY = 25;
    LARGE_BINARY = 26;

    TIME32 = 27;
    TIME64 = 28;
}`

and

`message ScalarValue{
    oneof value {

        (...)

        ScalarTime32Value time32_value = 30;
        ScalarTime64Value time64_value = 31;
        IntervalMonthDayNanoValue interval_month_day_nano = 32;
        StructValue struct_value = 33;
    }
}`

with the new messages:

`message ScalarTime32Value {
  oneof value {
    int32 time32_second_value = 1;
    int32 time32_millisecond_value = 2;
  };
}

message ScalarTime64Value {
  oneof value {
    int64 time64_microsecond_value = 1;
    int64 time64_nanosecond_value = 2;
  };
}`

Then, I would change  the `to_proto.rs` and `from_proto.rs` files consistently. What do you think, @isidentical, @alamb, @waitingkuo, @avantgardnerio?

Reply:
This change looks reasonable to me. It also corresponds to arrow's definition: https://github.com/apache/arrow/blob/master/format/Schema.fbs#L214-L231

Reply:
Thanks @waynexia! I will thus push my most up-to-date version where I implement these changes, with consistent modifications in `to_proto.rs` and `from_proto.rs`.

Reply:
I don't think we have worried about backwards compatibility in serialization before -- I recommend we simply make the protobuf definitions match the changes to `ScalarValue`s

Reply:
Thanks for the clarification, @alamb.